c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -4036,10 +4037,10 @@ bool fetch(JSContext *cx, unsigned argc, Value *vp) {
   }
 
   RootedString backend(cx, Request::backend(request));
-  if (!backend) {
+  if (!backend && builtins::Fastly::allowDynamicBackends == false) {
```

Comment:
What do you think about reordering these cases to one top-level check of `allowDynamicBackends`, with the different behavior for when `backend` is missing under those branches?